File: 48-rotate-image/rotate-image.cpp
class Solution {
private:
    void transposeMatrix(vector<vector<int>>& matrix) {
        for (int row = 0; row < matrix.size(); ++row) {
            for (int col = row; col < matrix[0].size(); ++col) {
                int temp = matrix[row][col];
                matrix[row][col] = matrix[col][row];
                matrix[col][row] = temp;
            }
        }
    }

    void reverseRowsInMatrix(vector<vector<int>>& matrix) {
        for (int r = 0; r < matrix.size(); ++r) {
            int left = 0;
            int right = matrix[0].size() - 1;

            while (left < right) {
                int temp = matrix[r][left];
                matrix[r][left] = matrix[r][right];
                matrix[r][right] = temp;

                ++left;
                --right;
            }
        }
    }

public:
    void rotate(vector<vector<int>>& matrix) {
        transposeMatrix(matrix);
        reverseRowsInMatrix(matrix);
    }
};

File: 48-rotate-image/rotate-image.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """

        ans = []
        n = len(matrix)
        ni = 0
        nj = 0
        for j in range(n):
            temp = []
            for i in range(n-1,-1,-1):
                temp.append(matrix[i][j])
            ans.append(temp)
        print(ans)

        for i in range(len(ans)):
            for j in range(len(ans)):
                matrix[i][j] = ans[i][j] 
                



        